services/jaeger/2.46.2/pre-upgrade/delete-jaeger-hr.yaml

```diff
@@ -0,0 +1,73 @@
+# For upgrades from DKP <2.6 to 2.6.x.
+# **REMOVE THIS AFTER 2.6 IS RELEASED**
+# Delete the jaeger helmrelease prior to upgrading.
+# This must be done because upgrading does not apply the newly added priorityClassName field on the
+# Jaeger spec, perhaps because the CRD has not been upgraded yet. Recreating the HelmRelease applies the
+# Jaeger spec properly. This job will only delete the v2.40.0 helmrelease (from 2.5), so rerunning this job
+# during upgrades within 2.6 will not result in the recreation of the helmrelease.
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: jaeger-pre-upgrade
+  namespace: ${releaseNamespace}
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: jaeger-pre-upgrade
+  namespace: ${releaseNamespace}
+rules:
+  - apiGroups: ["helm.toolkit.fluxcd.io"]
+    resources: ["helmreleases"]
+    verbs: ["get", "watch", "list", "delete"]
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: jaeger-pre-upgrade
+  namespace: ${releaseNamespace}
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: jaeger-pre-upgrade
+subjects:
+  - kind: ServiceAccount
+    name: jaeger-pre-upgrade
+    namespace: ${releaseNamespace}
+---
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: delete-jaeger-helmrelease
+  namespace: ${releaseNamespace}
+spec:
+  ttlSecondsAfterFinished: 100
+  template:
+    metadata:
+      name: delete-jaeger-helmrelease
+    spec:
+      serviceAccountName: jaeger-pre-upgrade
+      restartPolicy: OnFailure
+      priorityClassName: dkp-high-priority
+      containers:
+        - name: kubectl
+          image: bitnami/kubectl:1.26.4
+          command:
+            - sh
+            - "-c"
+            - |-
+              /bin/bash <<'EOF'
+              set -o nounset
+              set -o errexit
+              set -o pipefail
+
+              echo "checking jaeger helmrelease and deleting it pre-upgrade if v2.40.0"
+
+              VERSION=$(set -o errexit; kubectl get helmrelease -n ${releaseNamespace} jaeger --ignore-not-found -o=jsonpath='{.spec.chart.spec.version}')
+              echo "version $VERSION"
+              if [[ $VERSION == "2.40.0" ]]; then
```

Comment:
we likely won't bump this jaeger version in a 2.5 patch, but if we do then the upgrade from that patch version to 2.6.0 won't have the priority class set 🤔

Reply:
yeah, I don't think we really ever bump versions in patch releases unless there is a major bug; I think we should be safe here, jaeger is generally one of our more stable apps that we haven't really had any bug reports for. but to be safe, I guess I can make it trigger for anything less than 2.46.2

Reply:
looks good! thanks